src/nested_trait.jl

```diff
@@ -258,6 +258,27 @@ macro invoke_maker(argnum, type, sig)
 end
 
 
+macro next_trait_function(trait_type, sig)
```

Comment:
could we document this a bit?

Reply:
I've added some documentation

Reply:
I actually do not understand much of what is documented. That is fine with me, but one could add an example maybe?

Reply:
We have the mentioned file twice (in source and test) and in a not-so-detailed scrolling it was hard to find sections?

Reply:
How would you prefer to refer to that code?

Reply:
Hm, well, my main problem is, that I still do not directly see how to use it, if the macro is used within an eval (kind of like in a complicated environment overall. I would prefer a line like

```
Defining/using the Macro like

         @next_trait_function TraitList{IsEmbeddedManifold} f(M::AbstractDecoratorManifold, p, q)

bring the benefit, that you obtain more gummi bears... (or whatever the macro does)
```
Since I still have not yet understood the magic of this macro and the docs read like it basically does nothing but is still important for the compiler

Reply:
Well, it does nothing other than nicely asking the compiler to try harder optimizing some calls :wink: . It is not intended for normal users, because before attempting to use it you should check if it makes sense at all. The easiest way is by going through unoptimized Cthulhu output with remarks turned on, which is fairly hard to interpret. I think the only way someone other than me can use it is by adding a new trait or a new function to one of the two `@eval` blocks (either the one in ManifoldsBase.jl or a similar one in Manifolds.jl) and looking if it helps. Explaining the benefit here is fairly hard and maybe not worth the effort because hopefully some time in the future compiler won't need this kind of help.

Reply:
Ok, so my gummi bears was not that far away, you definitely need them once you have to use these ;)

But maybe a careful phrasing of your first sentence might help anyone here? Also the note that it affects certain traits and that there is one block at the end of the decorator_traits file where this is used. Maybe that would be a good choice to add these details? (and mention that its `src/decorator_traits.jl` at the end, to make the reference more precise).

Reply:
I've updated the docs a bit, feel free to suggest a change if you have an idea.

Reply:
OK, that's fine with me :)